dockerfiles/ci/buster/php-8.0/Dockerfile_shared_ext

```diff
@@ -0,0 +1,101 @@
+FROM datadog/dd-trace-ci:buster AS base
+
+ARG phpVersion
+ENV PHP_INSTALL_DIR_DEBUG_ZTS_ASAN=${PHP_INSTALL_DIR}/${phpVersion}-debug-zts-asan
+ENV PHP_INSTALL_DIR_DEBUG_NTS=${PHP_INSTALL_DIR}/${phpVersion}-debug
+ENV PHP_INSTALL_DIR_NTS=${PHP_INSTALL_DIR}/${phpVersion}
+ENV PHP_VERSION=${phpVersion}
+
+FROM base as build
+ARG phpTarGzUrl
+ARG phpSha256Hash
+RUN set -eux; \
+    curl -fsSL -o /tmp/php.tar.gz "${phpTarGzUrl}"; \
+    (echo "${phpSha256Hash} /tmp/php.tar.gz" | sha256sum -); \
+    tar xf /tmp/php.tar.gz -C "${PHP_SRC_DIR}" --strip-components=1; \
+    rm -f /tmp/php.tar.gz; \
+    cd ${PHP_SRC_DIR}; \
+    ./buildconf --force;
+COPY configure_shared_ext.sh /home/circleci/configure.sh
+
+FROM build as php-debug-zts-asan
+ENV CFLAGS='-fsanitize=address -DZEND_TRACK_ARENA_ALLOC'
+ENV LDFLAGS='-fsanitize=address'
+RUN set -eux; \
+    mkdir -p /tmp/build-php && cd /tmp/build-php; \
+    /home/circleci/configure.sh \
+        --enable-debug \
+        --enable-zts \
+        --prefix=${PHP_INSTALL_DIR_DEBUG_ZTS_ASAN} \
+        --with-config-file-path=${PHP_INSTALL_DIR_DEBUG_ZTS_ASAN} \
+        --with-config-file-scan-dir=${PHP_INSTALL_DIR_DEBUG_ZTS_ASAN}/conf.d; \
+    make -j "$((`nproc`+1))"; \
+    make install; \
+    mkdir -p ${PHP_INSTALL_DIR_DEBUG_ZTS_ASAN}/conf.d;
+
+FROM build as php-debug
+RUN set -eux; \
+    mkdir -p /tmp/build-php && cd /tmp/build-php; \
+    /home/circleci/configure.sh \
+        --enable-debug \
+        --prefix=${PHP_INSTALL_DIR_DEBUG_NTS} \
+        --with-config-file-path=${PHP_INSTALL_DIR_DEBUG_NTS} \
+        --with-config-file-scan-dir=${PHP_INSTALL_DIR_DEBUG_NTS}/conf.d; \
+    make -j "$((`nproc`+1))"; \
+    make install; \
+    mkdir -p ${PHP_INSTALL_DIR_DEBUG_NTS}/conf.d;
+
+FROM build as php-nts
+RUN set -eux; \
+    mkdir -p /tmp/build-php && cd /tmp/build-php; \
+    /home/circleci/configure.sh \
+        --prefix=${PHP_INSTALL_DIR_NTS} \
+        --with-config-file-path=${PHP_INSTALL_DIR_NTS} \
+        --with-config-file-scan-dir=${PHP_INSTALL_DIR_NTS}/conf.d; \
+    make -j "$((`nproc`+1))"; \
+    make install; \
+    mkdir -p ${PHP_INSTALL_DIR_NTS}/conf.d;
+
+FROM base as final
+COPY --chown=circleci:circleci --from=build $PHP_SRC_DIR $PHP_SRC_DIR
+COPY --chown=circleci:circleci --from=php-debug-zts-asan $PHP_INSTALL_DIR_DEBUG_ZTS_ASAN $PHP_INSTALL_DIR_DEBUG_ZTS_ASAN
+COPY --chown=circleci:circleci --from=php-debug $PHP_INSTALL_DIR_DEBUG_NTS $PHP_INSTALL_DIR_DEBUG_NTS
+COPY --chown=circleci:circleci --from=php-nts $PHP_INSTALL_DIR_NTS $PHP_INSTALL_DIR_NTS
+
+# Build core extensions as shared libraries.
+# We intentionally do not run 'make install' here so that we can test the
+# scenario where headers are not installed for the shared library.
+RUN set -eux; \
```

Comment:
Not required now, but in the future we could move it to an executable the installation, something like `php-install-ext-no-header pdo` so we do not have to rebuild containers at every new extension we want to test. Not required for this specific PR. If you find it useful can drop a few lines in a card, though.

Reply:
I'm not opposed to adding something like this, but I have a few questions:

1. Would `php-install-ext-no-header` be intended for use directly in CI? So we would add a new command to the job to build the extension every time?
2. How would the script handle extra configure flags? I'm asking because although `curl` and `pdo` have a standard build process, `pdo_mysql`, `pdo_pgsql`, and `pdo_sqlite` may require extra config options and/or `CFLAGS`. (I tried to install them here but I ran into trouble getting them to build since ext/pdo is shared.)

FWIW I added these to a separate layer at the very bottom of the Dockerfile so that if all the other layers are cached, this layer will build fairly quickly after adding a new extension. (I just timed adding ext/pcntl to this list and rebuilt the container and it took 42 seconds with cached layers.) But I also understand not wanting to have to rebuild the container if you haven't built it yet. :)

Reply:
> I'm asking because although curl and pdo have a standard build process, pdo_mysql, pdo_pgsql, and pdo_sqlite may require extra config options and/or CFLAGS

This alone can suggest that you made the right decision. I am resolving this comment, then. Thanks for claryfing.